packages/xdl/src/Analytics.ts

```diff
@@ -1,19 +1,115 @@
-import Segment from 'analytics-node';
+import SegmentAnalytics from 'analytics-node';
 import os from 'os';
 
-import { ip } from './internal';
+import { ConnectionType, ip } from './internal';
+
+// defaults are appended as comments
+type RedisOptions = {
+  port?: number; // 6379
+  host?: string; // 'localhost'
+  db?: number; // 0
+  password?: string;
+};
+
+// defaults are appended as comments
+type JobOptions = {
+  maxAttempts?: number; // 10
+};
+
+type QueueOptions = {
+  queueName?: string;
+  isMultiProcessor?: boolean;
+  prefix?: string;
+  redisOptions: RedisOptions;
+  jobOptions?: JobOptions;
+};
+
+type AnonymousId = {
+  anonymousId: string;
+};
+
+type UserId = {
+  userId: string;
+};
+
+type RudderStackId = AnonymousId | UserId;
+
+type RudderStackCommonUserTraits = {
+  userId: string;
+  currentConnection?: ConnectionType;
+  username?: string;
+  userType?: 'user' | 'robot';
+  primaryAccountId?: string;
+};
+
+type EventContext = {
+  ip: string;
+  device: {
+    model: string;
+    brand: string;
+  };
+  os: {
+    name: string;
+    version: string;
+  };
+  app?: { version?: string };
+};
+
+type TrackEvent = {
+  event: string;
+  properties?: any;
+  context?: EventContext;
+} & RudderStackId;
+
+type IdentifyEvent = {
+  traits?: RudderStackCommonUserTraits;
+  context?: EventContext;
+} & RudderStackId;
+
+interface RudderStackClient {
+  new (
+    writeKey: string,
+    dataPlaneURL: string,
+    options?: {
+      flushAt?: number; // queue size
+      flushInterval?: number; // ms
+      // the max number of elements that the SDK can hold in memory,
+      // this is different than the Redis list created when persistence is enabled
+      maxInternalQueueSize?: number;
+      logLevel?: string;
+    }
+  ): RudderStackClient;
+  createPersistenceQueue(
```

Comment:
add a comment to warn that this function is never called in client side code

Reply:
i've moved the typing to a d.ts file in the library